application/comit_node/src/settings/mod.rs

```diff
@@ -5,93 +5,191 @@ use crate::seed::Seed;
+            .parse()
+            .expect("comit_node listen address could not be parsed");
+        let btsieve_url =
+            Url::parse("http://localhost:8181").expect("Btsieve url could not be created");
+        let seed = Seed::new_random().expect("Could not generate random seed");
+
+        ComitNodeSettings {
+            comit: Comit { secret_seed: seed },
+            network: Network {
+                listen: vec![comit_listen],
+            },
+            http_api: HttpSocket {
+                address: IpAddr::V4(Ipv4Addr::UNSPECIFIED),
+                port: 8000,
+            },
+            btsieve: Btsieve {
+                url: btsieve_url,
+                bitcoin: PollParameters {
+                    poll_interval_secs: Duration::from_secs(300),
+                    network: "regtest".into(),
+                },
+                ethereum: PollParameters {
+                    poll_interval_secs: Duration::from_secs(20),
+                    network: "regtest".into(),
+                },
+            },
+            web_gui: Some(HttpSocket {
+                address: IpAddr::V4(Ipv4Addr::UNSPECIFIED),
+                port: 8080,
+            }),
+            log_levels: LogLevels {
+                comit_node: LevelFilter::Debug,
+            },
+        }
+    }
 }
 
 fn default_log() -> LevelFilter {
     LevelFilter::Debug
 }
 
-#[derive(Clone, Debug, Deserialize)]
+fn default_log_levels() -> LogLevels {
+    LogLevels {
+        comit_node: default_log(),
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct LogLevels {
+    #[serde(with = "self::serde_log", default = "default_log")]
+    pub comit_node: LevelFilter,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct Comit {
     pub secret_seed: Seed,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct Network {
     pub listen: Vec<Multiaddr>,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct HttpSocket {
     pub address: IpAddr,
     pub port: u16,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct Btsieve {
     #[serde(with = "url_serde")]
     pub url: url::Url,
     pub bitcoin: PollParameters,
     pub ethereum: PollParameters,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct PollParameters {
     #[serde(with = "self::serde_duration")]
     pub poll_interval_secs: Duration,
     pub network: String,
 }
 
 impl ComitNodeSettings {
-    pub fn create<D: AsRef<OsStr>, R: AsRef<OsStr>>(
-        default_config: D,
-        run_mode_config: R,
-    ) -> Result<Self, ConfigError> {
+    pub fn write_to(self, config_file: PathBuf) -> Result<Self, ConfigError> {
+        ComitNodeSettings::verify_directory_exists(&config_file)?;
```

Comment:
:upside_down_face: To make the side-effect clear.

```suggestion
        ComitNodeSettings::ensure_directory_exists(&config_file)?;
```